tools/checkpoint_quantization/quantize_llama3_checkpoints.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+export KERAS_BACKEND=jax CUDA_VISIBLE_DEVICES=
+"""
+
+import keras
+from absl import app
+from absl import flags
+
+import keras_nlp
+
+FLAGS = flags.FLAGS
+
+PRESETS = [
+    "llama3_8b_en",
+    "llama3_instruct_8b_en",
+]
+
+
+flags.DEFINE_string(
+    "preset",
+    None,
+    f'Must be one of {",".join(PRESETS)}',
+    required=True,
+)
+
+
+def quantize(preset):
+    model = keras_nlp.models.Llama3Backbone.from_preset(
+        preset, dtype="bfloat16"
+    )
+    model.quantize("int8")
+    return model
+
+
+def validate_output(model, tokenizer):
+    input_str = "What is Keras?"
+    length = 32
+
+    preprocessor = keras_nlp.models.Llama3CausalLMPreprocessor(tokenizer)
+    llama_lm = keras_nlp.models.Llama3CausalLM(
+        backbone=model, preprocessor=preprocessor
+    )
+    keras_output = llama_lm.generate([input_str], max_length=length)
+    keras_output = keras_output[0]
+    print("🔶 KerasNLP output:", keras_output)
+
+
+def main(_):
+    preset = FLAGS.preset
+    if preset not in PRESETS:
+        raise AssertionError(
+            f"Invalid preset {preset}. Must be one of {','.join(PRESETS)}"
+        )
+    print(f"🏃 Quantizing {preset}")
+
+    keras.config.set_floatx("bfloat16")
+
+    model = quantize(preset)
```

Comment:
We might be able to write this generally for all generative models, by using the `CausalLM` base class.

```python
causal_lm = keras_nlp.models.CausalLM(preset)
backbone = causal_lm.backbone
tokenizer = causal_lm.preprocessor.tokenizer
...
quantized_preset = preset + "_int8"
model.save_to_preset(quantized_preset)
tokenizer.save_to_preset(quantized_preset)
if FLAGS.upload_uri:
    keras_nlp.upload_preset(
        uri=FLAGS.upload_uri,
        quantized_preset,
    )
```

WDYT?

Reply:
This is nice! I will try it.